TakingITGlobal/commit2act
# [TIG-760] - Created Text Input Component with stories + testing

# Description
[TIG-760 - Task](https://app.clickup.com/t/9009201449/TIG-760)
[TIG-653 - Parent Task](https://app.clickup.com/t/9009201449/TIG-653)
[Related PR- Implementing i18n (Please look at this after this PR)]()

In this PR, I created a `TextInput` component from MUI's TextField Component and styled it accordingly to the [Figma](https://www.figma.com/file/pLQlafaAJPCvcsLHGbO1Gh/Commit2Act---Webapp?type=design&mode=design&t=3IH4WPa5ISOouUD7-0). 

This PR aims to create input components for text, email and password. This criteria includes:

- [ ] Can enter plain text, email address, or hidden password depending on the field type
- [ ] Can toggle hidden or visible password text
- [ ] Can be notified of invalid state if there is an error in the input value (using `error` prop)
- [ ] Can read a clear and descriptive label for each input
- [ ] Can be informed of input type, label, and error state when using screen reader

This PR does not implement the component TextInput in the signup flow.

# Testing
1. Pull down and install the branch (`npm install`)
2. Run all tests (`npm run test:basic`) and confirm they pass
3. Run Storybook (`npm run storybook`)
4. Make sure that you can type in all the `TextInput` input fields
5. Ensure `Outline Required` and `OutlinedPassword Required` have an error signal when they are empty (after being typed in)
6. Ensure `OutlinedPassword` can have the icon toggled on and off. When it is toggled on, the password should be visible and when it is toggled off, the password should not be visible.
7. Ensure `Outlined Email` have an error signal if user does not enter in a email address (containing @ and does not end in a '.')

# Screenshots
From Storybook:
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/05abfaf6-d5a1-4f0b-a3e3-0a7d83a923bd)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/d9fe5005-1a19-4fc9-ab19-26b25c9a3b8c)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/3ecb060c-478f-4422-985f-a4abfa19a1c3)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/9a84085c-3205-4078-94ab-36bc86562326)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/44210c23-707b-4662-a1f0-485bfd088c93)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/21fa8abd-d318-4d06-80ed-0d06908466be)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/734e64d1-6115-423e-aa66-ef56b3ceca08)

# EDIT: Code cleaned up  based on comment feedback!
